api/src/main/java/org/apache/cloudstack/query/QueryService.java

```diff
@@ -125,6 +125,10 @@ public interface QueryService {
     static final ConfigKey<Boolean> SharePublicTemplatesWithOtherDomains = new ConfigKey<>("Advanced", Boolean.class, "share.public.templates.with.other.domains", "true",
             "If false, templates of this domain will not show up in the list templates of other domains.", true, ConfigKey.Scope.Domain);
 
+    ConfigKey<Boolean> ReturnVmStatsOnVmList = new ConfigKey<>("Advanced", Boolean.class, "return.vm.stats.on.vm.list", "true",
```

Comment:
```suggestion
    ConfigKey<Boolean> ReturnVmStatsOnVmList = new ConfigKey<>("Advanced", Boolean.class, "return.vm.stats.on.vm.list", "false",
```

After due considerations, I think this can be set to false what originally Joao wants to accomplish. As the listVirtualMachines API is used by at least three integrations I know of (CAPC, k8s-provider and terraform etc.) and several portals are built on top such APIs, it makes sense to trade off speed. BUT, if we do this we must document this change in the release notes and what users can do to get previous/old behaviour if they want it.

What do you think - @JoaoJandre @weizhouapache @DaanHoogland @sureshanaparti @shwstppr ?

Reply:
obviously users/other projects will benifit from the change, meanwhile it will break the backward compatibility. 
if we agree more pros than cons, let's go ahead. 
as @rohityadavcloud said, we must document in the release note and let users know (although many users might not be impacted and not be aware of the change).

Reply:
We add the config for users who may want the backward compatibility. But we use an appropriate default that benefits most users.